### PR TITLE
Rail 4165 add support for adding attribute filters

### PR DESCRIPTION
**Pull request by MilanLempera**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8272/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10123/ : FAILURE in 8m 44s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8273/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10124/ : FAILURE in 8m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8287/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10142/ : FAILURE in 5m 29s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8288/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10143/ : FAILURE in 5m 03s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8290/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10145/ : FAILURE in 5m 25s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts [hunk 1]
```diff
@@ -12,13 +12,15 @@ export async function canFilterBeAdded(
         return true;
     }
 
-    const [addedDisplayForm, ...existingDisplayForms] = await ctx.backend
+    const addedDisplayForm = await ctx.backend
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts:15]: You could run those two in parallel making it a bit faster.

File: libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts [hunk 2]
```diff
@@ -128,6 +128,15 @@ export const selectIsExport = createSelector(selectConfig, (state) => {
     return state.isExport;
 });
 
+/**
+ * Returns whether the Dashboard is white labeled.
+ *
+ * @internal
+ */
+export const selectIsWhiteLabeled = createSelector(selectConfig, (state) => {
+    return state.isWhiteLabeled;
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts:137]: Should we add a default value here? Something like `?? false`?

File: libs/sdk-ui-dashboard/src/model/types/commonTypes.ts [hunk 3]
```diff
@@ -124,6 +124,12 @@ export interface DashboardConfig {
      */
     isExport?: boolean;
 
+    /**
+     * @internal
+     * When dashboard is white labeled, we hide GD links and branding.
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/types/commonTypes.ts:129]: Nitpick: please put the text above the internal specifier.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx [hunk 4]
```diff
@@ -102,23 +107,52 @@ export function DefaultFilterBar(props: IFilterBarProps): JSX.Element {
...
+                            onAttributeFilterAdded={addAttributeFilterPlaceholder}
+                        />
+                    );
+                }
             })}
+            <AttributeFilterDropZone
+                targetIndex={attributeFiltersCount}
+                onDrop={addAttributeFilterPlaceholder}
+            />
+            <div style={{ position: "relative", flexGrow: 1, height: "55px" }}>
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx:148]: Should this be a class?

File: libs/sdk-ui-dashboard/src/presentation/dragAndDrop/draggableAttributeFilter/DraggableAttributeFilterPlaceholder.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// (C) 2022 GoodData Corporation
+import React from "react";
+import { useDashboardDrag } from "../useDashboardDrag";
+import classNames from "classnames";
+import { useDashboardSelector } from "../../../model";
+import { selectIsInEditMode } from "../../../model/store/ui/uiSelectors";
+
+export type CustomDashboardAttributeFilterPlaceholderComponentProps = {
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/dragAndDrop/draggableAttributeFilter/DraggableAttributeFilterPlaceholder.tsx:8]: Should these be in a types.ts file instead? Do they need some TSDoc? Or are they just some internal thing?

File: libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts (hunk 2 above)
Comment (MilanLempera) [reply libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts:137]: It is not necessary, default value is set in [initializeDashboardHandler/resolveDashboardConfig.ts](https://github.com/gooddata/gooddata-ui-sdk/pull/2613#diff-d11e08af68868d9ccf444ce9827c7a8a0270faa1dbd8fbf91d8697f1361103ef)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8291/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10147/ : FAILURE in 5m 35s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8295/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10151/ : FAILURE in 5m 37s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8298/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10155/ : SUCCESS in 16m 17s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2613

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2380/ : SUCCESS in 1m 01s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2674/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1492/ : SUCCESS in 17m 48s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10156/ : SUCCESS in 15m 53s

**Pull request merged**